### DISCUSSION TITLE
Self-assisted Troubleshooting (Check before open issues)

### DISCUSSION THREAD

**Discussion by oasiscifr** [trojan-gfw/igniter#373] [Announcements and Readmes]
If igniter didn't work as you expected, take steps below to narrow the range of problems.

### 0 Use `adb` to obtain the log

```
adb logcat Trojan-TUN-lwIP:V TrojanConfig:V Trojan:V trojan:V igniter:V Clash:V ClashConfig:V ProxyService:V Go:V GoLog:V VPN:V VpnService:V libc:V DEBUG:V *:S >> igniter-log.txt
```
Please upload `igniter-log.txt` file

### 1.1 Tap the "Test Connection" button.
If it said "Connected ...", then congratulations, your server as well as the igniter is working. The problem only happened to some specific domains. Go to 2.1.

If it said "Connection closed by peer", go to 1.2.

### 1.2 Check the server status and what you filled in igniter
The most common problems here:
* the server is using self-signed certificates, but you turned on the "Verify Certificate"
* you filled IP rather than a domain in the "Remote Address" field.
* the server is down
* the server is wrongly configured
* the server is inaccessible through TCP
* the server is behind a CDN service
Try to solve it by yourself. You can open a new issue **ONLY IF** eliminated all problems above.

### 2.1 Why some specific Apps/ Websites didn't work
* Stop VPN -> Turn off "Exempt Chinese Domains/IP" -> make sure your App is not in "Exempt Apps" list  -> Start VPN ->  
turn on Airplane Mode of your phone -> wait 10 seconds -> turn off  Airplane Mode

By doing this, all traffic should be surely redirected to the server.

Check your App/website again. If your problem persists, go to 2.2

### 2.2 The server can not access some specific domains
The most common problems here:
* The system DNS of the **SERVER** is also polluted, try to modify your system DNS (for example. add `1.1.1.1` to the first line of `/etc/resolv.conf`)

For the problem listed without a solution, which means it has nothing to do with the client(igniter), you should solve it by yourself.

You can open a new issue if all problems above were excluded.

**Comment by TchaikovDriver** [comment]
### 2.3 Igniter didn't work in Google Play Store
Try the following instructions:
- Make sure the version of Igniter is v0.9.6 or higher.
- Igniter homepage -> Exempt Apps
- Choose `Allow Mode`, check all the applications whose names contain _android_, _google_, _system_. Save and exit.
Start Igniter and check the issue again.

**EDIT by wongsyrone**: You may clear the app data of Google Play and Google Play Service and try again.

**Comment by noobww** [comment]
In 2.2, what if my device doesn't have a "resolv.conf" in /etc/ ...😂

**Comment by noobww** [comment]
> ### 2.3 Igniter didn't work in Google Play Store
> Try the following instructions:
> 
> * Make sure the version of Igniter is v0.9.6 or higher.
> * Igniter homepage -> Exempt Apps
> * Choose `Allow Mode`, check all the applications whose names contain _android_, _google_, _system_. Save and exit.
>   Start Igniter and check the issue again.
> 
> **EDIT by wongsyrone**: You may clear the app data of Google Play and Google Play Service and try again.

This works for me!!
